include/bluetooth/mesh/access.h

```diff
@@ -256,27 +256,48 @@ struct bt_mesh_model_op {
  */
 #define BT_MESH_PUB_TRANSMIT_INT(transmit) ((((transmit) >> 3) + 1) * 50)
 
+/** Model publication context. */
 struct bt_mesh_model_pub {
-	/* Self-reference for easy lookup */
+	/** The model the context belongs to. Initialized by the stack. */
 	struct bt_mesh_model *mod;
 
-	u16_t addr;         /* Publish Address */
-	u16_t key;          /* Publish AppKey Index */
-
-	u8_t  ttl;          /* Publish Time to Live */
-	u8_t  retransmit;   /* Retransmit Count & Interval Steps */
-	u8_t  period;       /* Publish Period */
-	u8_t  period_div:4, /* Divisor for the Period */
-	      cred:1,       /* Friendship Credentials Flag */
-	      count:3;      /* Retransmissions left */
-
-	/* Buffer containing the publication message */
+	u16_t addr;         /**< Publish Address. */
+	u16_t key;          /**< Publish AppKey Index. */
+
+	u8_t  ttl;          /**< Publish Time to Live. */
+	u8_t  retransmit;   /**< Retransmit Count & Interval Steps. */
+	u8_t  period;       /**< Publish Period. */
+	u8_t  period_div:4, /**< Divisor for the Period. */
+	      cred:1,       /**< Friendship Credentials Flag. */
+	      count:3;      /**< Retransmissions left. */
+
+	u32_t period_start; /**< Start of the current period. */
+
+	/** @brief Buffer containing the publication message.
+	 *
+	 *  The application is expected to initialize this with
+	 *  a valid net_buf_simple pointer, with the help of e.g.
+	 *  the NET_BUF_SIMPLE() macro. The buffer must contain
+	 *  a valid publication messages before calling the
+	 *  bt_mesh_model_publish() API or after the publication's
+	 *  update() callback has been called and returned success.
+	 */
 	struct net_buf_simple *msg;
 
-	/* Publish callback */
-	void    (*func)(struct bt_mesh_model *mod);
-
-	/* Publish Period Timer */
+	/** @brief Callback for updating the publication buffer.
+	 *
+	 *  When set to NULL, the model is assumed not to support
+	 *  periodic publishing. When set to non-NULL the callback
+	 *  will be called periodically and is expected to update
+	 *  the publication buffer with a valid publication message.
```

Comment:
Maybe refer explicitly here to bt_mesh_model_pub :: msg

Reply:
Fixed now, however due to conflicts I ended up fixing this as a new patch rather than amending the existing one. I hope that's ok.